src/main/java/software/amazon/msk/auth/iam/internals/MSKCredentialProvider.java

```diff
@@ -311,8 +328,9 @@ else if (externalId != null) {
 
         STSAssumeRoleSessionCredentialsProvider createSTSRoleCredentialProvider(String roleArn,
                                                                                 String sessionName, String stsRegion) {
+            EndpointConfiguration endpointConfiguration = buildEndpointConfiguration(stsRegion);
             AWSSecurityTokenService stsClient = AWSSecurityTokenServiceClientBuilder.standard()
-                    .withRegion(stsRegion)
+                    .withEndpointConfiguration(endpointConfiguration)
```

Comment:
Do we need to set the endpoint configuration? if the stsRegion is set to `aws-global` it should be using `sts.amazonaws.com` anyway.

> regional – The SDK or tool always uses the AWS STS endpoint for the currently configured Region. For example, if the client is configured to use us-west-2, all calls to AWS STS are made to the Regional endpoint sts.us-west-2.amazonaws.com, instead of the global sts.amazonaws.com endpoint. To send a request to the global endpoint while this setting is enabled, you can set the Region to aws-global.

https://docs.aws.amazon.com/sdkref/latest/guide/feature-sts-regionalized-endpoints.html

Have you tested this out? In KDA you should be able to define an environment variable and override the REGION config.

My concern with manually setting endpointconfiguration is the following endpoint might not be valid for all aws regions and partitions in the future.
```
EndpointConfiguration(
                            String.format("sts.%s.amazonaws.com", stsRegion),
                            stsRegion);
```

Reply:
I published a new commit to use region.getServiceEndpoint API instead of a string.

Reply:
Could you also try setting region as aws-global in Flink?

Reply:
aws-global is not reachable for customer who are running KDA or MSK Connect in a private subnet. They have to use a regional endpoint for STS.